st2common/st2common/models/api/action.py

```diff
@@ -148,6 +149,11 @@ class ActionAPI(BaseAPI):
                 "patternProperties": {
                     "^\w+$": util_schema.get_draft_schema()
                 }
+            },
+            "tags": {
+                "description": "User associated metadata assigned to this object.",
```

Comment:
I have deliberately tried to keep the scope of this property narrow to have only key-value tags. The use cases we have are identity, grouping etc so staying away from adding in more complex objects. In theory though this can be extended to arbitrary metadata with values as complex objects although I am not yet certain we have identified a usecase for supporting that ability.